test-suite/shared/src/test/scala/org/scalajs/testsuite/javalib/io/CharArrayReaderTest.scala

```diff
@@ -0,0 +1,183 @@
+  @Test def test_Constructor$C(): Unit = {
+    cr = new CharArrayReader(hw)
+    assertTrue("Failed to create reader", cr.ready)
+  }
+
+  /**
+    * @tests java.io.CharArrayReader#CharArrayReader(char[], int, int)
+    */
+  @throws[IOException]
+  @Test def test_Constructor$CII(): Unit = {
+    cr = new CharArrayReader(hw, 5, 5)
+    assertTrue("Failed to create reader", cr.ready)
+    val c = cr.read
+    assertTrue("Created incorrect reader--returned '" + c.toChar + "' intsead of 'W'", c == 'W')
+  }
+
+  /**
+    * @tests java.io.CharArrayReader#close()
+    */
+  @Test def test_close(): Unit = {
+    cr = new CharArrayReader(hw)
+    cr.close()
+    try {
+      cr.read
+      fail("Failed to throw exception on read from closed stream")
+    } catch {
+      case e: IOException =>
+
+      // Expected
+    }
+    // No-op
+    cr.close()
+  }
+
+  /**
+    * @tests java.io.CharArrayReader#mark(int)
+    */
+  @throws[IOException]
+  @Test def test_markI(): Unit = {
+    cr = new CharArrayReader(hw)
+    cr.skip(5L)
+    cr.mark(100)
+    cr.read
+    cr.reset()
+    assertEquals("Failed to mark correct position", 'W', cr.read)
+  }
+
+  /**
+    * @tests java.io.CharArrayReader#markSupported()
+    */
+  @Test def test_markSupported(): Unit = {
+    cr = new CharArrayReader(hw)
+    assertTrue("markSupported returned false", cr.markSupported)
+  }
+
+  /**
+    * @tests java.io.CharArrayReader#read()
+    */
+  @throws[IOException]
+  @Test def test_read(): Unit = {
+    cr = new CharArrayReader(hw)
+    assertEquals("Read returned incorrect char", 'H', cr.read)
+    cr = new CharArrayReader(Array[Char]('\u8765'))
+    assertTrue("Incorrect double byte char", cr.read == '\u8765')
+  }
+
+  /**
+    * @tests java.io.CharArrayReader#read(char[], int, int)
+    */
+  @throws[IOException]
+  @Test def test_read$CII(): Unit = {
+    val c = new Array[Char](11)
+    cr = new CharArrayReader(hw)
+    cr.read(c, 1, 10)
+    assertTrue("Read returned incorrect chars", new String(c, 1, 10) == new String(hw, 0, 10))
+  }
+
+  /**
+    * @tests java.io.CharArrayReader#ready()
+    */
+  @throws[IOException]
+  @Test def test_ready(): Unit = {
+    cr = new CharArrayReader(hw)
+    assertTrue("ready returned false", cr.ready)
+    cr.skip(1000)
+    assertTrue("ready returned true", !cr.ready)
+    cr.close()
+    try {
+      cr.ready
+      fail("No exception 1")
+    } catch {
+      case e: IOException =>
+
+      // expected
+    }
+    try {
+      cr = new CharArrayReader(hw)
+      cr.close()
+      cr.ready
+      fail("No exception 2")
+    } catch {
+      case e: IOException =>
+
+    }
+  }
+
+  /**
+    * @tests java.io.CharArrayReader#reset()
+    */
+  @throws[IOException]
+  @Test def test_reset(): Unit = {
+    cr = new CharArrayReader(hw)
+    cr.skip(5L)
+    cr.mark(100)
+    cr.read
+    cr.reset()
+    assertEquals("Reset failed to return to marker position", 'W', cr.read)
+    // Regression for HARMONY-4357
```

Comment:
What is this testing?

Reply:
moved to a separate test and indicated the testing around the original charbuffer offset